x-pack/plugins/security_solution/server/plugin.ts

```diff
@@ -289,6 +289,19 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
         logger: this.logger,
         cache: this.exceptionsCache,
       });
+
+      // initialize the global agnostic Trusted Apps list
+      exceptionListClient
+        .createTrustedAppsList()
+        .then((response) => {
+          if (response !== null) {
+            this.logger.info('Trusted Apps list was created');
+          }
+        })
+        .catch((error) => {
+          this.logger.error('Failed to create Trusted Apps list via list plugin');
+          this.logger.error(error);
+        });
```

Comment:
In the past we have been told that on startup of Kibana as a plugin the best way to do things was to be careful about auto adding data to any parts of Kibana before the user has ever clicked on a particular plugin or make it to where if the end user is very concerned about data storage they are allowed to pretty much "clean slate things" by removing data within indexes including the .kibana saved objects index and if they don't click the plugin again they will not get that data added.

I think this had to do with worries that we and other plugins would begin ballooning up Kibana indexes with data on startup and some people might be paying some type of cost for their disk storage, so we wanted to keep the amount of data provisioning low until users began clicking within features.

So we have been careful about auto provisioning data on startup. I would say some good examples is that beats does not auto-create Kibana dashboards, indexes, ILM rules, etc... until you download a beat and then call setup on it to push kibana dashboards, indexes, etc... into your instance.

For detections (as another example) we rely on the users to touch the detections page and then we provision and create any missing detection structures based on the user credentials of that first user entering the detections page. This gives us an opportunity to give warnings to the end user that there are some permission issues and that someone with appropriate administration abilities needs to create the structures by visiting the page. It also gives us the opportunity to have the initially created data structures tied to users with permissions for down the road if we decide to do more with types of ownership, RBAC, etc... 

Outside of that, the other reasoning has been that auto-provisioning and creating data on startup would force us to use a system level user which in this case looks to be 'kibana'. In this example here, when you call `.createTrustedAppsList()` it looks like the user that is going to be shown to have created the list initially will be 'kibana'.

So any user interfaces that first show the list, will show the user who created the list as being "kibana"? In our current RBAC type system level with saved objects what permissions does Kibana have? I really don't know and do not know how the RBAC saved object system is going to work with a list that is created with the user of Kibana. I might be missing something here as I am new to system level accounts creating things within Kibana myself.

This might be ok? I don't know, but previously we were under guidance that user interfaces showing system level users as owning things or having created things were not good things as the user would begin to become concerned that they don't know who "Kibana" is and why "Kibana" is creating data structures instead of their authorized and authenticated users on their system since their system could be wired into active directory (for example) and they do not have a user named Kibana.

Reply:
If we decide to leave this, I think one of us should swap out the `kibana` level user which is reserved for `kibana_system` as the user since `kibana` looks deprecated:

<img width="1321" alt="Screen Shot 2020-08-12 at 5 08 19 PM" src="https://user-images.githubusercontent.com/1151048/90077161-ae09db80-dcbe-11ea-9486-fef235f8704a.png">

Reply:
Thanks @FrankHassanabad for the history here. I'm surprised to learn that even the `.kibana` index should not be populated until someone in the UI actually visits the plugin. I have seen issues in the ingest side with that approach and it just feels wrong especially if we are enabling plugins by default in Kibana.

When I first read your comment I thought that perhaps we would have an issue in 7.10 if this list is not created in order to ensure that the Endpoint Polices are updated with a `manifest` entry for Trusted Apps (even if the content of the artifact is empty), but after a quick look at the code, I think that will be ok and the actual `elastic-endpoint.exe` seems to also handle the case when it will not be in the manifest, so I think we're good there.